flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/LocalInputChannel.java

```diff
@@ -95,7 +95,7 @@ public LocalInputChannel(
 	// ------------------------------------------------------------------------
```

Comment:
Please copy PR description into the commit message and once you have some expected benchmark results, please include them in the commit message as well :) 

Could you also explain in both places, how are you solving the problem? Maybe something along those lines:

```
Before this change, in network benchmark (for example 1000 channels benchmark with 4 record writers) StreamNetworkBenchmarkEnvironment#createInputGate was creating 1000 input gates with 4 input channels each, which doesn't make much sense. This commit is changing that to a single receiver with 4 input gates, with 1000 channels each.

It is achieved by providing testing implementations of InputChannels, which are using channel index for requesting subpartitions from ResultPartition, instead of subpartition index. Thanks to that, we can connect single ResultPartition with N subpartitions, to a single instance of InputGate with 1000 channels.
```
?